Fix incorrect default transform values on foreign 3D nodes

SaracenOne opened the pull request:
Closes #56434
Introduces property_can_revert and property_get_revert methods to Node3D to work around the problem described in the issue. Does this by getting the default value for "transform" then converting it to the desired editor property.

SaracenOne committed:
Fix incorrect default transform values on foreign 3D nodes

RandomShaper commented on scene/3d/node_3d.cpp:
```diff
@@ -845,6 +846,84 @@ void Node3D::_validate_property(PropertyInfo &property) const {
+	if (p_name == "basis") {
+		Variant variant = PropertyUtils::get_property_default_value(this, "transform", &valid);
+		if (valid) {
+			if (variant.get_type() == Variant::Type::TRANSFORM3D) {
```
Suggestion: make these checks more compact by writing them as `if (valid && variant.get_type() == Variant::Type::...)` so that only one `else` branch is needed.

RandomShaper reviewed (approved):
Looks fine.

JFonS reviewed (approved):
Looks good to me.

akien-mga commented on scene/3d/node_3d.cpp:
```diff
@@ -916,6 +995,9 @@ void Node3D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("to_local", "global_point"), &Node3D::to_local);
 	ClassDB::bind_method(D_METHOD("to_global", "local_point"), &Node3D::to_global);
 
+	ClassDB::bind_method(D_METHOD("property_can_revert"), &Node3D::property_can_revert);
+	ClassDB::bind_method(D_METHOD("property_get_revert"), &Node3D::property_get_revert);
```
You need to define the argument name:
```suggestion
	ClassDB::bind_method(D_METHOD("property_can_revert", "name"), &Node3D::property_can_revert);
	ClassDB::bind_method(D_METHOD("property_get_revert", "name"), &Node3D::property_get_revert);
```

SaracenOne commented:
Updated with suggestion from @RandomShaper to make the if/else chain more compact, and added parameter name as @akien-mga mentioned which should make this PR pass the test suite now

akien-mga commented:
Thanks!
